tt_metal/CMakeLists.txt

```diff
@@ -18,7 +18,7 @@ set(TT_METAL_OBJECTS
 
 add_library(tt_metal ${TT_METAL_OBJECTS})
 if(BUILD_SHARED_LIBS)
-    target_link_libraries(tt_metal PUBLIC device)
+    target_link_libraries(tt_metal PUBLIC device metal_common_libs)
```

Comment:
@vtangTT This is crazy - we didn't need these libs for metal? Wow - guess everything is taken care of by umd